### ISSUE TITLE
Loss Nan

### ISSUE THREAD

**Issue by Lature-Yang** [dvlab-research/Stratified-Transformer#26]
Hi, I'm reproducing your excellent work, the  loss is reached to the nan with  obtained 0 mIou.  Is the learning rate set incorrectly？

aug: True
base_lr: 0.006
batch_size: 8
batch_size_test: 4
batch_size_val: 4
channels: [48, 96, 192, 384]
classes: 13
concat_xyz: True
data_name: s3dis
data_root: ../../Dataset/S3DIS/trainval_fullarea
depths: [2, 2, 6, 2]
dist_backend: nccl
dist_url: tcp://127.0.0.1:58501
distributed: True
downsample_scale: 8
drop_path_rate: 0.3
drop_rate: 0.5
epochs: 50
eval_freq: 1
evaluate: True
fea_dim: 6
grid_size: 0.04
grid_sizes: [0.04, 0.08, 0.16, 0.32]
ignore_label: 255
jitter_clip: 0.02
jitter_sigma: 0.005
k: 16
loop: 30
manual_seed: 123
max_batch_points: 140000
max_num_neighbors: 34
model_path: None
momentum: 0.9
multiplier: 0.1
multiprocessing_distributed: True
names_path: data/s3dis/s3dis_names.txt
ngpus_per_node: 4
num_heads: [3, 6, 12, 24]
num_layers: 4
optimizer: AdamW
patch_size: 0.04
print_freq: 1
quant_size: 0.01
quant_sizes: [0.01, 0.02, 0.04, 0.08]
rank: 0
ratio: 0.25
rel_key: True
rel_query: True
rel_value: True
resume: exp/s3dis/stratified_transformer/model/model_last.pth
save_folder: None
save_freq: 1
save_path: exp/s3dis/stratified_transformer/model
scheduler: MultiStep
scheduler_update: epoch
split: val
start_epoch: 0
stem_transformer: True
step_epoch: 30
sync_bn: True
test_area: 5
test_gpu: [0]
test_list: data/s3dis/list/val5.txt
test_list_full: data/s3dis/list/val5_full.txt
test_workers: 4
train_gpu: [0, 1, 2, 3]
transformer_lr_scale: 0.1
up_k: 3
use_amp: True
use_xyz: True
voxel_max: 80000
voxel_size: 0.04
warmup: linear
warmup_iters: 1500
warmup_ratio: 1e-06
weight: None
weight_decay: 0.01
window_size: [0.16, 0.32, 0.64, 1.28]
workers: 4
world_size: 4

[32m[06/02 00:16:07 main-logger]: [0mTrain result at epoch [23/50]: mIoU/mAcc/allAcc 0.0130/0.0769/0.1693.
WARNING:root:NaN or Inf found in input tensor.
[32m[06/02 00:16:07 main-logger]: [0m>>>>>>>>>>>>>>>> Start Evaluation >>>>>>>>>>>>>>>>
[32m[06/02 00:16:28 main-logger]: [0mTest: [1/17] Data 4.412 (4.412) Batch 21.110 (21.110) Loss nan (nan) Accuracy 0.2793.
[32m[06/02 00:16:31 main-logger]: [0mTest: [2/17] Data 0.002 (2.207) Batch 3.324 (12.217) Loss nan (nan) Accuracy 0.1809.
[32m[06/02 00:16:36 main-logger]: [0mTest: [3/17] Data 0.001 (1.472) Batch 4.249 (9.561) Loss nan (nan) Accuracy 0.1969.
[32m[06/02 00:16:38 main-logger]: [0mTest: [4/17] Data 0.001 (1.104) Batch 2.364 (7.762) Loss nan (nan) Accuracy 0.1815.
[32m[06/02 00:16:48 main-logger]: [0mTest: [5/17] Data 0.001 (0.883) Batch 10.526 (8.314) Loss nan (nan) Accuracy 0.1848.
[32m[06/02 00:17:02 main-logger]: [0mTest: [6/17] Data 0.001 (0.736) Batch 13.099 (9.112) Loss nan (nan) Accuracy 0.1507.
[32m[06/02 00:17:08 main-logger]: [0mTest: [7/17] Data 0.001 (0.631) Batch 5.935 (8.658) Loss nan (nan) Accuracy 0.1931.
[32m[06/02 00:17:09 main-logger]: [0mTest: [8/17] Data 0.001 (0.553) Batch 1.564 (7.771) Loss nan (nan) Accuracy 0.1636.
[32m[06/02 00:17:10 main-logger]: [0mTest: [9/17] Data 0.001 (0.491) Batch 1.386 (7.062) Loss nan (nan) Accuracy 0.1597.
[32m[06/02 00:17:13 main-logger]: [0mTest: [10/17] Data 0.000 (0.442) Batch 2.654 (6.621) Loss nan (nan) Accuracy 0.1734.
[32m[06/02 00:17:15 main-logger]: [0mTest: [11/17] Data 0.001 (0.402) Batch 2.049 (6.205) Loss nan (nan) Accuracy 0.2362.
[32m[06/02 00:17:26 main-logger]: [0mTest: [12/17] Data 0.001 (0.369) Batch 11.297 (6.630) Loss nan (nan) Accuracy 0.1980.
[32m[06/02 00:17:29 main-logger]: [0mTest: [13/17] Data 0.002 (0.340) Batch 2.617 (6.321) Loss nan (nan) Accuracy 0.1651.
[32m[06/02 00:17:55 main-logger]: [0mTest: [14/17] Data 0.001 (0.316) Batch 25.620 (7.700) Loss nan (nan) Accuracy 0.1804.
[32m[06/02 00:18:04 main-logger]: [0mTest: [15/17] Data 0.001 (0.295) Batch 9.624 (7.828) Loss nan (nan) Accuracy 0.1890.
[32m[06/02 00:18:06 main-logger]: [0mTest: [16/17] Data 0.001 (0.277) Batch 1.992 (7.463) Loss nan (nan) Accuracy 0.1893.
[32m[06/02 00:18:09 main-logger]: [0mTest: [17/17] Data 0.001 (0.260) Batch 2.720 (7.184) Loss nan (nan) Accuracy 0.1934.
[32m[06/02 00:18:10 main-logger]: [0mVal result: mIoU/mAcc/allAcc 0.0146/0.0769/0.1903.
[32m[06/02 00:18:10 main-logger]: [0mClass_0 Result: iou/accuracy 0.1903/1.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_1 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_2 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_3 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_4 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_5 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_6 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_7 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_8 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_9 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_10 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_11 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0mClass_12 Result: iou/accuracy 0.0000/0.0000.
[32m[06/02 00:18:10 main-logger]: [0m<<<<<<<<<<<<<<<<< End Evaluation <<<<<<<<<<<<<<<<<
WARNING:root:NaN or Inf found in input tensor.
[32m[06/02 00:18:10 main-logger]: [0mSaving checkpoint to: exp/s3dis/stratified_transformer/model/model_last.pth
[32m[06/02 00:18:10 main-logger]: [0mlr: [0.006, 0.0006000000000000001]
batch_size shortened from 2 to 1, points from 156591 to 80000
[32m[06/02 00:18:20 main-logger]: [0mEpoch: [24/50][1/765] Data 4.474 (4.474) Batch 10.469 (10.469) Remain 60:03:51 Loss nan Lr: [0.006, 0.0006] Accuracy 0.1705.

**Comment by X-Lai** [comment]
It's weird. I never met this problem before. I guess maybe different cuda versions or types of gpus cause NaN. Can you try to disable amp by simply setting the `use_amp` to `False`?

**Comment by Lature-Yang** [comment]
Hi, I trained some epochs again. The issue phenomenon still exists,  however, the miou did not grow steadily, and even declined after I remove the amp. Here are part of the traning log.
My enviroment: CUDA-10.2, gcc-7.5, Pytorch-1.9.0.

https://drive.google.com/drive/folders/16Dbqi0Ji8jSGqzowiPEF83nTTfXPjuw3?usp=sharing

**Comment by X-Lai** [comment]
It seems like I need permission to access the google doc @Lature-Yang

**Comment by Lature-Yang** [comment]
> It seems like I need permission to access the google doc @Lature-Yang

My mistake. I modify the link：https://pan.baidu.com/s/1wnZiLvqYp1HhhQHl168b6w 
提取码：o6e2

**Comment by Lature-Yang** [comment]
> It seems like I need permission to access the google doc @Lature-Yang

Hi, do you have any resolutions about this issue?

**Comment by X-Lai** [comment]
@Lature-Yang 
Sorry for the delay. I have checked your config and it is correct. But I observe that some classes yield all zeros accuracy in the validation result. I think you'd better check whether your data preparation is correct.

**Comment by X-Lai** [comment]
Please strictly follow the guide of S3DIS preprocessing in the repo https://github.com/yanx27/Pointnet_Pointnet2_pytorch.

**Comment by Lature-Yang** [comment]
Hi，I solved the problem by uninstalling the environment and reinstall it. It's so weird. The obtained is up to 69.8% ， while your trained model can  even reach to 72.4%.

**Comment by X-Lai** [comment]
@Lature-Yang  Have you tested the model using `test.py`? If not, please follow the README.md file to test the model. Note that the `validate` in `train.py` is different from `test.py`, because it does not reproject the voxelized points back to all points.

**Comment by Lature-Yang** [comment]
Surely, the best validation mIou is 68.08%. I used it to test, and the Miou reached 69.8%. Here is the traninig log.
[log.txt](https://github.com/dvlab-research/Stratified-Transformer/files/8940913/log.txt)

**Comment by X-Lai** [comment]
@Lature-Yang I have checked your config and I find that you use 2 GPUs but still maintain the batch size to 8. I think it is weird to produce the result. Moreover, one strange thing is that you only need about 56 hours for training with only 2 GPUs, but I need 70 hours for training with 4 2080Ti GPUs. I know that you may use 3090 or more advanced GPUs, but the gap of training time is somehow too large. I recommend you check whether the training batch size is really 8 or not. 

I retrain the model previously and the training log is here [log.txt](https://github.com/dvlab-research/Stratified-Transformer/files/8946896/log.txt). FYI, there is also another issue (https://github.com/dvlab-research/Stratified-Transformer/issues/12) where the result can be successfully reproduced.

**Comment by Buanma** [comment]
Hi, I have a question that I am very much looking forward to getting an answer. In the training result I found that the iou of Class_3 is always 0, is this normal? 
As shown below:
![image](https://github.com/dvlab-research/Stratified-Transformer/assets/92349662/e0f4ffed-1dc8-4eb9-99e3-49d76b8453cb)
Thanks a lot !
